### PR TITLE
Documentation about custom data collectors with autowire/autoconfigure

### LINKED ISSUE #14060: [FrameworkBundle] Allow to leverage autoconfiguration for DataCollector…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#37332
| PR author(s) | @l-vo
| Merged in    | 5.2-dev

### PR DESCRIPTION
**Pull request by l-vo**
Feature integrated in symfony/symfony#37332

Close #14060

### CODE REVIEW

File: profiler/data_collector.rst [hunk 1]
```diff
@@ -289,3 +293,12 @@ The position of each panel in the toolbar is determined by the collector priorit
...
 Most built-in collectors use ``255`` as their priority. If you want your collector
 to be displayed before them, use a higher value (like 300).
+
+.. versionadded:: 5.2
+
+    :class:`Symfony\\Bundle\\FrameworkBundle\\DataCollector\\AbstractDataCollector` was introduced in Symfony 5.2.
+
+.. note::
+
+    Before the introduction of :class:`Symfony\\Bundle\\FrameworkBundle\\DataCollector\\AbstractDataCollector`, template path was defined in the service configuration (`template` key). This is still possible to define the template in the service configuration. In this case **template in service configuration takes precedence over template definied in data collector code**.
```
Comment (stof) [inline profiler/data_collector.rst:303]: typo. `defined`

Comment (l-vo) [reply profiler/data_collector.rst]: Fixed, thank you :)

**Comment by javiereguiluz** [comment]
Laurent, thanks for contributing these docs and for implementing the original feature too.

While merging we decided to reword parts of the original article, so we did some changes to your contribution too. Most of it was left intact, but we changed some contents to always use the new AbstractDataCollector, which makes things much simpler.

**Comment by l-vo** [comment]
Great 🙂  You're welcome !

**Pull request merged**